inventory/osinfo/osinfo_linux.go

```diff
@@ -104,24 +103,22 @@ func GetDistributionInfo() (*DistributionInfo, error) {
 	case util.Exists(rhRelease):
 		releaseFile = rhRelease
 		parseReleaseFunc = parseEnterpriseRelease
-	default:
-		err = errors.New("unable to obtain release info, no known /etc/*-release exists")
 	}
 
 	b, err := ioutil.ReadFile(releaseFile)
 	if err != nil {
-		di = &DistributionInfo{ShortName: Linux}
+		oi = &OSInfo{ShortName: Linux}
 	} else {
-		di = parseReleaseFunc(string(b))
+		oi = parseReleaseFunc(string(b))
 	}
 
-	out, err := getUname()
-	if err != nil {
-		return nil, err
+	var uts unix.Utsname
+	if err := unix.Uname(&uts); err != nil {
+		return nil, fmt.Errorf("unix.Uname error: %v", err)
 	}
-	di.Kernel = strings.TrimSpace(string(out))
-	// No need to get fancy here, assume the binary architecture
-	// is the same as the system.
-	di.Architecture = Architecture(runtime.GOARCH)
-	return di, nil
+	oi.Hostname = string(uts.Nodename[:])
```

Comment:
i don't quite understand the purpose of this line; what do you get out of unix.Uname that needs this treatment

Reply:
This is a byte array, [65]byte. Better than what it is in syscall [65]int8